packages/styled-react/src/components/ActionList.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {forwardRef} from 'react'
+import styled from 'styled-components'
+import {
+  ActionList as PrimerActionList,
+  type ActionListProps as PrimerActionListProps,
+  type ActionListItemProps as PrimerActionListItemProps,
+  type ActionListLinkItemProps as PrimerActionListLinkItemProps,
+  type ActionListGroupProps as PrimerActionListGroupProps,
+  type ActionListDividerProps as PrimerActionListDividerProps,
+  type ActionListLeadingVisualProps as PrimerActionListLeadingVisualProps,
+  type ActionListTrailingVisualProps as PrimerActionListTrailingVisualProps,
+} from '@primer/react'
+import {sx, type SxProp} from '../sx'
+import type {ForwardRefComponent} from '../polymorphic'
+import {Box} from './Box'
+
+type PrimerActionListTrailingActionProps = React.ComponentProps<typeof PrimerActionList.TrailingAction>
+
+export type ActionListProps<As extends React.ElementType = 'ul'> = PrimerActionListProps<As> & SxProp
+export type ActionListItemProps = React.PropsWithChildren<PrimerActionListItemProps & SxProp>
+export type ActionListLinkItemProps = PrimerActionListLinkItemProps & SxProp
+export type ActionListGroupProps = React.PropsWithChildren<PrimerActionListGroupProps & SxProp>
+export type ActionListDividerProps = React.PropsWithChildren<PrimerActionListDividerProps & SxProp>
+export type ActionListLeadingVisualProps = React.PropsWithChildren<PrimerActionListLeadingVisualProps & SxProp>
+export type ActionListTrailingVisualProps = React.PropsWithChildren<PrimerActionListTrailingVisualProps & SxProp>
+export type ActionListTrailingActionProps = PrimerActionListTrailingActionProps & SxProp
+
+const StyledActionList = styled(PrimerActionList).withConfig({
+  shouldForwardProp: (prop: string | number) => prop !== 'sx',
+})`
+  ${sx}
+` as typeof PrimerActionList & {
+  <As extends React.ElementType = 'ul'>(props: ActionListProps<As>): React.ReactElement | null
+}
+
+const ActionListImpl = React.forwardRef(function ActionListImpl<As extends React.ElementType = 'ul'>(
+  {as, ...rest}: ActionListProps<As>,
+  ref: React.ComponentPropsWithRef<As>['ref'],
+) {
+  return <StyledActionList ref={ref} {...rest} {...(as ? {forwardedAs: as} : {})} />
+})
```

Comment:
let's make sure to do this for the other subcomponents if they support the as prop as well 🙏